dll/win32/syssetup/install.c

```diff
@@ -531,17 +541,108 @@ StatusMessageWindowProc(
     IN WPARAM wParam,
     IN LPARAM lParam)
 {
+    PDLG_DATA pDlgData;
     UNREFERENCED_PARAMETER(wParam);
 
+    pDlgData = (PDLG_DATA)GetWindowLongPtrW(hwndDlg, GWLP_USERDATA);
+
     switch (uMsg)
     {
         case WM_INITDIALOG:
         {
+            BITMAP bm;
             WCHAR szMsg[256];
 
+            /* Init dialog data */
+            pDlgData = HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, sizeof(*pDlgData));
+            if (pDlgData)
+            {
+                SetWindowLongPtrW(hwndDlg, GWLP_USERDATA, (LONG_PTR)pDlgData);
+            }
+
+            /* Load bitmaps */
+            pDlgData->hLogoBitmap = LoadImageW(hDllInstance,
+                                                MAKEINTRESOURCEW(IDB_REACTOS), IMAGE_BITMAP,
+                                                0, 0, LR_DEFAULTCOLOR);
+
+            pDlgData->hBarBitmap = LoadImageW(hDllInstance, MAKEINTRESOURCEW(IDB_LINE),
+                                            IMAGE_BITMAP, 0, 0, LR_DEFAULTCOLOR);
+            GetObject(pDlgData->hBarBitmap, sizeof(bm), &bm);
+            pDlgData->BarWidth = bm.bmWidth;
+            pDlgData->BarHeight = bm.bmHeight;
+
+            if (pDlgData->hLogoBitmap && pDlgData->hBarBitmap)
+            {
+                if (SetTimer(hwndDlg, IDT_BAR, 20, NULL) == 0)
+                {
+                    DPRINT1("SetTimer(IDT_BAR) failed: %d\n", GetLastError());
+                }
+
+                /* Get the animation bar control */
+                pDlgData->hWndBarCtrl = GetDlgItem(hwndDlg, IDC_BAR);
+            }
+
+            /* Get and set status text */
             if (!LoadStringW(hDllInstance, IDS_STATUS_INSTALL_DEV, szMsg, ARRAYSIZE(szMsg)))
                 return FALSE;
             SetDlgItemTextW(hwndDlg, IDC_STATUSLABEL, szMsg);
+
+            return TRUE;
+        }
+
+        case WM_TIMER:
+        {
+            if (pDlgData->hBarBitmap)
```

Comment:
If allocating `pDlgData` failed we'll have null here as well. Perhaps `if (uMsg != WM_INITDIALOG && pDlgData == NULL) return FALSE;` or some such at the top, to avoid having to add null checks in each individual case?